nixos/modules/services/x11/display-managers/sddm.nix

```diff
@@ -14,7 +14,7 @@ let
   xserverWrapper = pkgs.writeScript "xserver-wrapper" ''
     #!/bin/sh
     ${concatMapStrings (n: "export ${n}=\"${getAttr n xEnv}\"\n") (attrNames xEnv)}
-    exec ${dmcfg.xserverBin} ${dmcfg.xserverArgs} "$@"
+    exec systemd-cat ${dmcfg.xserverBin} ${dmcfg.xserverArgs} "$@"
```

Comment:
Just tested this -- logging consistently works  with journal restarts, stop and start breaks logging till display-manager is restarted (which seems expected and compared to what we have now is better anyway).